modules/java/native/java_machine.c

```diff
@@ -194,6 +195,9 @@ _setup_jvm_options_array(JavaVMSingleton *self, const gchar *jvm_options_str)
                                                 g_strdup_printf("-Djava.library.path=%s",
                                                     resolvedConfigurablePaths.initial_module_path));
 
+  jvm_options_array = _jvm_options_array_append(jvm_options_array,
+                                                g_strdup_printf("-Dlog4j.configurationFactory=org.syslog_ng.logging.CustomConfigurationFactory"));
```

Comment:
Is it possible to override this accidentally with custom JVM options?

Reply:
No, that's why we have `predefined_options` to prevent that.